replica/database.cc

```diff
@@ -1028,6 +1028,13 @@ future<> database::drop_column_family(const sstring& ks_name, const sstring& cf_
     f.get(); // re-throw exception from truncate() if any
 }
 
+future<> database::drop_table_on_all(sstring ks_name, sstring cf_name, timestamp_func tsf, bool with_snapshot) {
```

Comment:
A better name is drop_table_on_all_shards(), so people don't have to wonder on all what (invoke_on_all is a sharded<> method, so that provides some context).

Another way to go is static database::drop_table_on_all_shards(sharded<database>&), which makes it even clearer by the type of the first parameter.

Reply:
sure, will do.

Reply:
Using becomes quite long and cumbersome, e.g.
`replica::database::drop_table_on_all_shards(e.db(), etc...)`

How about shortening it to `replica::database::drop_table(e.db(), etc...)`
The caller shouldn't care how it's done internally in the `sharded<database>`, does it matter to the caller if we decided to implement it e.g. only on shard 0, or on the local shard?

Reply:
Changed to a static `drop_table_on_all_shards` method in v4.

Reply:
Long it better than short when things are not as expected (operating on the sharded container rather than the object called on)